metadata-db-app/src/cmr/metadata_db/data/oracle/concepts.clj

```diff
@@ -410,28 +423,31 @@
   [db provider concept]
   (try
     (j/with-db-transaction
-     [conn db]
-     (if-let [error (or (validate-concept-id-native-id-not-changing db provider concept)
-                        (when (= :variable-association (:concept-type concept))
-                          (or (validate-same-provider-variable-association concept)
-                              (validate-collection-not-associated-with-another-variable-with-same-name db concept))))]
+      [conn db]
+      (if-let [error (or (validate-concept-id-native-id-not-changing db provider concept)
+                         (when (= :variable-association (:concept-type concept))
+                           (or (validate-same-provider-variable-association concept)
+                               (validate-collection-not-associated-with-another-variable-with-same-name db concept))))]
        ;; There was a concept id, native id mismatch with earlier concepts
-       error
+        error
        ;; Concept id native id pair was valid
-       (let [{:keys [concept-type]} concept
-             table (tables/get-table-name provider concept-type)
-             seq-name (str table "_seq")
-             [cols values] (concept->insert-args concept (:small provider))
-             stmt (format (str "INSERT INTO %s (id, %s, transaction_id) VALUES "
-                               "(%s.NEXTVAL,%s,GLOBAL_TRANSACTION_ID_SEQ.NEXTVAL)")
-                          table
-                          (string/join "," cols)
-                          seq-name
-                          (string/join "," (repeat (count values) "?")))]
-         (trace "Executing" stmt "with values" (pr-str values))
-         (j/db-do-prepared db stmt values)
-         (after-save conn provider concept)
-         nil)))
+        (let [{:keys [concept-type]} concept
+              table (tables/get-table-name provider concept-type)
+              seq-name (str table "_seq")
+              [cols values] (concept->insert-args concept (:small provider))
+              stmt (format (str "INSERT INTO %s (id, %s, transaction_id) VALUES "
+                                "(%s.NEXTVAL,%s,GLOBAL_TRANSACTION_ID_SEQ.NEXTVAL)")
+                           table
+                           (string/join "," cols)
+                           seq-name
+                           (string/join "," (repeat (count values) "?")))]
+          (trace "Executing" stmt "with values" (pr-str values))
+          (when (not (= "efs-only" efs-config/efs-toggle))
+            (j/db-do-prepared db stmt values)
+            (after-save conn provider concept))
+          (when (not (= "efs-off" efs-config/efs-toggle))
+            (efs/save-concept provider concept-type concept))
```

Comment:
All that EFS is being used for is storing the metadata element. Some things, like concept-id and revision-id might be in EFS as a result of the file path, but other metadata about the metadata is going to be stored in a different system(an as of yet undetermined system. DynamoDB is on the list to consider). For the save function, I have the entire concept being passed in because there are elements that will need to be pulled from it to determine where to store it in EFS